rest/admin_api.go

```diff
@@ -181,7 +181,24 @@ func getAuthScopeHandleCreateDB(ctx context.Context, h *handler) (bucketName str
 		Bucket string `json:"bucket"`
 	}
 	reader := bytes.NewReader(bodyJSON)
-	err = DecodeAndSanitiseConfig(ctx, reader, &dbConfigBody, false)
+
+	// read and decode json to pull bucket name from the config
+	b, err := io.ReadAll(reader)
+	if err != nil {
+		return "", err
+	}
+
+	// Expand environment variables if needed
+	if base.BoolDefault(h.server.Config.Unsupported.AllowDbConfigEnvVars, true) {
+		b, err = expandEnv(h.ctx(), b)
```

Comment:
We're doing bytes->reader->bytes for no reason here with NewReader->ReadAll 

We should be able to pass `bodyJSON` straight into `expandEnv` and `ConvertBackQuotedStrings`, since they don't modify the slice, they return a modified copy of it.

It does feels like we need to push those two pieces (`expandEnv`/`ConvertBackQuotedStrings`) into a shared function called by this, `DecodeAndSanitiseConfig`, and `readSanitizeJSON`.

We can pass a bool in to that function control env var behaviour and just have it take bytes and return bytes.